### DISCUSSION TITLE
Querying for CustomFieldTypes

### DISCUSSION THREAD

**Discussion by rotkiv1205** [jesper-raemaekers/python-polarion#118] [Q&A] [answered]
Hi people!

In my project I want fetch all the current CustomFieldTypes. I thought it would be a good idea to use the following:
defined_data_types = client.getService('Tracker').getDefinedCustomFieldTypes('...', '...')

However, this returns an error saying that it didn't anticipate 'enumID' and that it's supposed to be 'ID'.
Any ideas of why that would be? Also, i don't find much documentation on this so if there are, i'd gladly look through it! :D

**Comment by jesper-raemaekers** [comment]
You're close. So it would want a project and a workitem type, like this:
```
defined_data_types = client.getService('Tracker').getDefinedCustomFieldTypes('project','testcase')
```

**Reply by rotkiv1205** [reply]
When trying with your suggestion the problem still persists.
![image](https://github.com/jesper-raemaekers/python-polarion/assets/37670665/54fdf7cc-a99a-47c5-aa7c-a4ed6426b127)

Also, as to not cause confusion, this is my client.
client = Polarion(adress, user, password)

**Comment by rotkiv1205** [comment, marked as answer]
After some research and thinking I found `.getAllEnumOptionsForId()` which provided the expected output for me. This would probably explain why i got the error message since I used it for the wrong purpose. Thanks for the reply.
